tests/vtep/tests.c

```diff
@@ -126,6 +154,236 @@ static struct ether_addr multicast_get_dst_addr(uint32_t ip)
 	return dst;
 }
 
+static inline struct ether_addr
+multicast_get_dst6_addr(union pg_ipv6_addr ip)
+{
+	struct ether_addr dst;
+
+	/* Forge dst mac addr */
+	dst.addr_bytes[0] = 0x33;
+	dst.addr_bytes[1] = 0x33;
+	dst.addr_bytes[2] = ip.word8[12];
+	dst.addr_bytes[3] = ip.word8[13];
+	dst.addr_bytes[4] = ip.word8[14];
+	dst.addr_bytes[5] = ip.word8[15];
+	return dst;
+}
+
+static void test_vtep6_simple(void)
+{
+	struct ether_addr mac_dest = {{0xb0, 0xb1, 0xb2,
+				      0xb3, 0xb4, 0xb5} };
+	struct ether_addr mac_src = {{0xc0, 0xc1, 0xc2,
+				      0xc3, 0xc4, 0xc5} };
+	struct ether_addr multicast_mac1, multicast_mac2;
+	/*we're usin the same cfg for the collect and the hub*/
+	struct pg_brick *vtep_west, *vtep_east;
+	struct pg_brick *collect_west1, *collect_east1;
+	struct pg_brick *collect_west2, *collect_east2;
+	struct pg_brick *collect_hub, *hub;
+	struct pg_graph *graph;
+	struct rte_mempool *mp = pg_get_mempool();
+	struct rte_mbuf *pkts[NB_PKTS];
+	struct rte_mbuf **result_pkts;
+	struct pg_error *error = NULL;
+	struct ether_hdr *tmp;
+	union pg_ipv6_addr ip_src = {.word64 = {0,1}};
+	union pg_ipv6_addr ip_dst = {.word64 = {0,2}}; 
+	union pg_ipv6_addr ip_vni1 = {.word16 = {0xff,0,0,0,0,0,0,0x1}};
+	union pg_ipv6_addr ip_vni2 = {.word16 = {0xff,0,0,0,0,0,0,0x2}};
+	int flag = 0;
+	uint16_t i;
+
+	/*For testing purpose this vtep ip is 1*/
+	vtep_west = pg_vtep6_new("encapsulatron", 500, PG_EAST_SIDE,
```

Comment:
Well, encapsulatron is the brick name, "encapsulation" is not really a name, encaplulator might be a better name ?